src/helpers/clipboard/clipboard_figure_state.ts

```diff
@@ -100,3 +106,60 @@ export class ClipboardFigureState implements ClipboardState {
 
   drawClipboard(renderingContext: GridRenderingContext): void {}
 }
+
+export class ClipboardFigureChart {
+  private readonly copiedChart: AbstractChart;
+
+  constructor(
+    private dispatch: CommandDispatcher["dispatch"],
+    getters: Getters,
+    readonly sheetId: UID,
+    copiedFigureId: string
+  ) {
+    const chart = getters.getChart(copiedFigureId);
+    if (!chart) {
+      throw new Error(`No chart for the given id: ${copiedFigureId}`);
+    }
+    this.copiedChart = chart.copyInSheetId(sheetId);
+  }
+
+  paste(sheetId: UID, position: { x: number; y: number }, size: FigureSize) {
+    const copy = this.copiedChart.copyInSheetId(sheetId);
+    const newId = new UuidGenerator().uuidv4();
+    this.dispatch("CREATE_CHART", {
+      id: newId,
+      sheetId,
+      position,
+      size: size,
+      definition: copy.getDefinition(),
+    });
+    return newId;
+  }
+}
+
+export class ClipboardFigureImage {
+  private readonly copiedImage: ImageDefinition;
+
+  constructor(
+    private dispatch: CommandDispatcher["dispatch"],
+    getters: Getters,
+    readonly sheetId: UID,
+    copiedFigureId: string
+  ) {
+    const image = getters.getImage(copiedFigureId);
+    this.copiedImage = deepCopy(image);
+  }
+
+  paste(sheetId: UID, position: { x: number; y: number }, size: FigureSize) {
+    const copy = deepCopy(this.copiedImage);
```

Comment:
Image don't need any deepCopy since we can not edit the image definition for now. But by principle, we need to save the state of the figure when we copy it and we can paste any number of copy has we want without wanting those copy to have the same definition.